pkg/skaffold/build/cache/hash.go

```diff
@@ -80,14 +105,19 @@ func getHashForArtifact(ctx context.Context, depLister DependencyLister, a *late
 	if args != nil {
 		inputs = append(inputs, args...)
 	}
+	return encode(inputs)
+}
 
+func encode(inputs []string) (string, error) {
+	if len(inputs) == 0 {
+		return "", nil
```

Comment:
Is this what was produced previously if there were no inputs?  Or would the Encode() error?
(Would we receive `0000000000000000000` or something similar, and is this something we don't want to support?)

Reply:
Previously these lines were embedded in another function and it wasn't possible to get nil inputs since at least the `Artifact` struct was marshalled and added to it. But I agree that it should probably error. Fixed.